src/cascadia/TerminalApp/JsonUtils.h

```diff
@@ -427,6 +501,35 @@ namespace TerminalApp::JsonUtils
             return AllClear;
         }
 
+        Json::Value ToJson(const T& val)
+        {
+            if (val == AllClear)
+            {
+                return BaseEnumMapper::ToJson(AllClear);
+            }
+            else if (val == AllSet)
+            {
+                return BaseEnumMapper::ToJson(AllSet);
+            }
+            else if (WI_IsSingleFlagSet(val))
+            {
+                return BaseEnumMapper::ToJson(val);
+            }
+            else
+            {
+                Json::Value json{ Json::ValueType::arrayValue };
+                for (const auto& pair : TBase::mappings)
+                {
+                    if (pair.second != AllClear &&
+                        (val & pair.second) == pair.second)

```

Comment:
use `WI_IsFlagSet`

Reply:
oh we can't, can we. because they may not have defined the flag operations?

but wait, you used one above. hm

Reply:
☹ This is that thing where wil complains about the flag not being a "compile-time constant expression"

<details>

```
1>D:\Terminal\src\cascadia\TerminalApp\JsonUtils.h(524,1): error C2975: 'flag': invalid template argument for 'wil::details::verify_single_flag_helper', expected compile-time constant expression (compiling source file JsonUtilsTests.cpp)
1>D:\Terminal\dep\wil\include\wil\Common.h(564): message : see declaration of 'flag' (compiling source file JsonUtilsTests.cpp)
1>D:\Terminal\src\cascadia\TerminalApp\JsonUtils.h(505): message : while compiling class template member function 'Json::Value TerminalApp::JsonUtils::FlagMapper<JsonTestFlags,TerminalApp::JsonUtils::ConversionTrait<JsonTestFlags>>::ToJson(const T &)'
1>        with
1>        [
1>            T=JsonTestFlags
1>        ] (compiling source file JsonUtilsTests.cpp)
1>D:\Terminal\src\cascadia\TerminalApp\JsonUtils.h(686): message : see reference to function template instantiation 'Json::Value TerminalApp::JsonUtils::FlagMapper<JsonTestFlags,TerminalApp::JsonUtils::ConversionTrait<JsonTestFlags>>::ToJson(const T &)' being compiled
1>        with
1>        [
1>            T=JsonTestFlags
1>        ] (compiling source file JsonUtilsTests.cpp)
1>D:\Terminal\src\cascadia\ut_app\JsonUtilsTests.cpp(92): message : see reference to class template instantiation 'TerminalApp::JsonUtils::FlagMapper<JsonTestFlags,TerminalApp::JsonUtils::ConversionTrait<JsonTestFlags>>' being compiled
```

</details>

Reply:
Meh. Resolving.